account_banking_mandate/models/account_banking_mandate.py

```diff
@@ -145,6 +145,14 @@ def create(self, vals=None):
             vals['unique_mandate_reference'] = \
                 self.env['ir.sequence'].next_by_code(
                     'account.banking.mandate') or 'New'
+        if self._context.get('mandate_bank_partner_view'):
+            partner_id = self._context['partner_id']
+            acc_num = self.env.context.get('acc_number')
+            partner_bank_id = self.env['res.partner.bank'].search([
+                ('partner_id', '=', partner_id),
+                ('acc_number', '=', acc_num)], limit=1)
```

Comment:
Why not passing directly the res.partner.bank record ID and remove this ugly search?

Reply:
This was the main problem of this module. The res.partner.bank is not properly passed through the view. Instead of passing the `active_id` of the res.partner.bank, it passes the id of the res.partner belonging to that res.partner.bank.
As mandates are created through a one2many with this approach the job is done correctly.

Reply:
Is this a one2many? Then the problem is that you are showing the partner_bank_id field in it. If not shown, then all is handled automatically by Odoo.

Reply:
I see your point but partner_bank_id it is shown: https://github.com/OCA/bank-payment/blob/11.0/account_banking_mandate/views/account_banking_mandate_view.xml#L34

Reply:
Then create a specific embedded view for this one and everything solved without this ugly code.